### PR TITLE
Add support for Seq2Seq LMs

### PR DESCRIPTION
**Pull request by Vinno97**
Hi,

This PR adds support for seq2seq (encoder-decoder) models like T5, UL2 and their FLAN variants.
The FLAN models are pretty good few/zero shot learners and found it was a shame we couldn't use them here.

About the 'Not a "catch all"', I don't know for certain that all encoder-decoder models contain this key in their config. The ones I checked do. However, most decoder models also contain a "is-decoder" key, except GPT-2.

I kin of just merged the support in there, but don't think I could do much better without rewriting stuff.

Let me know what you guys think.

### CODE REVIEW

**Comment by Ph0rk0z** [comment]
Maybe I can finally use that reddit model that has been taking space on my HDD.

**Comment by oobabooga** [comment]
I have tested it with `google/flan-t5-xl` and it worked. It worked even in chat mode. Thank you!

File: modules/models.py [hunk 1]
```diff
@@ -55,7 +56,12 @@ def find_model_type(model_name):
     elif any((k in model_name for k in ['gpt4chan', 'gpt-4chan'])):
         return 'gpt4chan'
     else:
-        return 'HF_generic'
+        config = AutoConfig.from_pretrained(f"{shared.args.model_dir}/{model_name}")
```
Comment (aaroncoffey) [inline modules/models.py:59]: It appears that model_dir or model_name already has a `/`. This is breaking the install for me. Consider os.path.join for this type of concatenation.

Comment (oobabooga) [reply modules/models.py:59]: I missed that, should be fixed now https://github.com/oobabooga/text-generation-webui/commit/9c2e7c0fab385c8208a17a19feaa43cacf44bcf1

**Comment by Ph0rk0z** [comment]
Finally https://huggingface.co/hyunwoongko/reddit-9B works.

**Comment by evangineer** [comment]
Fixes #1357

**Pull request merged**